### ISSUE TITLE
How to know user spent time on page

### ISSUE THREAD

**Issue by iwindfree** [akamai/boomerang#345]
Hi, I use boomerang script.

I want to know  the amount of time a user spends on a page. 
In boomerang document, 
" `rt.quit`: This parameter will exist (but have no value) if the beacon was fired as part of the `onbeforeunload` event. This is typically used to  find out how much time the user spent on the page before leaving "

For page unload beacon(rt.quit exist) , what value I used to find out how long user stayed in page.
I kept checking the source, but I couldn't find out exactly, so I'm asking.

**Comment by bluesmoon** [comment]
Use the t_done value from that beacon

**Comment by iwindfree** [comment]
Thanks for your reply.
but  is t_done is page elapsed time? 
In my test, for example.
I entered the page1. ( 10:00:00)
I leaved the page1. (10:00:30)  // after 30 seconds
When I leaved the page1, I received page unload beacon (rt.quit exist).
but in page unload beacon, t_done value is 270ms.
I think that it may not the time user stayed in that page.

However, it can take a lot of time in a place with a lot of data, so we asked if we could know the corresponding value from the unload beacon.

I think that I could  find the time which user stayed a page  with the same session and the same page id in the history data and compare the values. However, it can take a lot of time in  a lot of data, so I  asked if we could know the corresponding value from the unload beacon. (above example, 30 sec)

Thanks in advance.

**Comment by bluesmoon** [comment]
May be a bug in `t_done` then. In any case, you can look at `rt.end` and subtract either `rt.tstart` or `nt_load_end` depending on whether you want it from navigation start or load complete.

**Comment by iwindfree** [comment]
Thanks for your reply.

But unload beacon (rt.quit exist), rt.end - rt.start not represent page stay time. It only equal to t_done.
I think that I search the first beacon and unload beacon for same page id, and rt.end of unload beacon and substract rt.tstart or nt_load_end of first beacon.

**Comment by bluesmoon** [comment]
Unless you have a SPA, nt_load_end and rt.tstart are always the timestamp of the navigation start of the page. They will be the same for all beacons on the same navigation. For single page apps there are multiple navigations in a single page, so maybe that is what you're seeing.

**Comment by iwindfree** [comment]
Thnaks for your reply.
I build boomerang script with spa plugin support SPA based web page and normal web page.
In my test web page is not SPA type. but boomerang is buid with spa plugin, the problem happened. 
I try to another way to check page stayed time. thank you.

**Comment by nicjansma** [comment]
@iwindfree you're correct that we're not necessarily adding a separate "now" timestamp on the Unload beacon.

The parameters like `rt.end` and `rt.tstart` etc. are for the Page Load itself, not the "now" of the Unload beacon.

It's a good idea, as it could track total page view time for that particular page load.  We could add a `rt.unload` or something similar.

**Comment by bluesmoon** [comment]
I think `rt.end` should be the timestamp when the beacon is sent (regardless of beacon type). We use that to calculate total session duration.
